### PR TITLE
YPP Dashboard

### LINKED ISSUE #3151: YT Dashboard
## Context

- https://github.com/joystream/atlas/issues/2973

## Scope

- [ ] Verified status
- [ ] Tier display
- [ ] Programme Details Widget
- [ ] Discord Widget
- [ ] Payments Widget 
> widget links TBC 
- [ ] Rewardable actions (based on channel tier) 
> (re)query Youtube on page load?
- [ ] Multiple channels info box (bottom of the page)

## Design 

[Link to Figma> ](https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1805%3A133265)

Screenshot 2022-08-18 at 21 19 50

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #3151 

**Design**
https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1805%3A133265

Sorry for fixing NftTile here but I noticed it was crashing in storybook

### CODE REVIEW

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            iconPlacement: 'right',
+          }}
+        />
+      </WidgetsWrapper>
+      <RewardsWrapper>
+        {REWARDS.map((reward) => (
+          <BenefitCard key={reward.title} {...reward} joyAmount={new BN(1200000000000)} />
+        ))}
+      </RewardsWrapper>
+      {/* TODO: add tooltip once it will show up on design */}
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: image

this icon is not for tooltips(check other banners in the app). It is just an icon and you didn't add this.

File: packages/atlas/src/components/WidgetTile/WidgetTile.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}) => {
+  const mdMatch = useMediaMatch('md')
+  const lgMatch = useMediaMatch('lg')
+  return (
+    <Wrapper>
+      <Title hasTooltip={!!tooltip}>
+        <Text variant="h100" as="p" color="colorText">
+          {title}
+        </Text>
+        {tooltip && <Information {...tooltip} />}
```
Comment (drillprop) [inline packages/atlas/src/components/WidgetTile/WidgetTile.tsx:41]: This is probably the Information component issue, but the gap between the label and the header is too big, when the Information component is present: 
image
image

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.styled.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { css } from '@emotion/react'
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.styled.ts]: You have a typo in the file name `YppDashboard.styled.ts` - should be `YppDashboard.styles.ts`

File: packages/atlas/src/components/WidgetTile/WidgetTile.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { SkeletonLoader } from '@/components/_loaders/SkeletonLoader'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+
+import { Content, StyledButton, TextWrapper, Title, Wrapper } from './WidgetTile.styles'
+
+export type WidgetTileProps = {
+  loading?: boolean
+  title: string | number
+  text: string | number
+  caption?: string | number
```
Comment (drillprop) [inline packages/atlas/src/components/WidgetTile/WidgetTile.tsx]: I'm not sure if we should put here string/number. What if we want to use here our NumberFormat component?

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.styled.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  ${media.sm} {
+    grid-template-columns: repeat(2, auto);
+  }
+`
+
+export const TierCount = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${sizes(1)};
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.styled.ts]: I think we still should avoid using gap property with flex

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    actionButton: { text: 'Get referral link' },
+    steps: [
+      'Copy your link with get referral link button',
+      'Send it to the person who you want to recommend',
+      'You will be rewarded for every new successful sign up, that uses your referral link to land on the YPP page before signing up',
+    ],
+  },
+]
+
+const TIERS = {
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: We could use an array here, right?

**Review by drillprop — CHANGES_REQUESTED** [review]
Found a few minor issues, besides those, look really solid 👍🏻

**Review by drillprop — APPROVED** [review]

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+  TierWrapper,
+  WidgetsWrapper,
+} from '@/views/global/YppDashboard/YppDashboard.styles'
+
+const REWARDS = [
+  {
+    title: 'Publish video on Atlas',
+    description:
+      'Publishing your existing content on Atlas is the fastest way to earn more JOY tokens. Simply follow those steps:',
+    actionButton: { text: 'Publish new video' },
```
Comment (kdembler) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: This should open video workspace

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+          the same as used previously on YouTube
+        </Text>
+      </>,
+      'Your uploads will be periodically reviewed for quality and completeness. Videos that match the criteria of programme participation listed above will score towards overall rewards',
+    ],
+  },
+  {
+    title: 'Share Atlas video on YouTube',
+    description: 'To share Atlas video you need to first upload your own video to the platform.',
+    actionButton: { text: 'Go to My videos' },
```
Comment (kdembler) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: This should navigate to my videos

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 9]
```diff
@@ -0,0 +1,174 @@
+import BN from 'bn.js'
```
Comment (kdembler) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: Dashboard shouldn't be global, it should be only available in studio. For now, let's have YPP in viewer link to landing page and in studio, let's have it link to the dashboard, we'll update later

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+import {
+  Header,
+  RewardsWrapper,
+  StyledSvgAlertsInformative24,
+  TierCount,
+  TierDescription,
+  TierWrapper,
+  WidgetsWrapper,
+} from '@/views/global/YppDashboard/YppDashboard.styles'
```
Comment (kdembler) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: This should be relative import

File: packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+const REWARDS = [
```
Comment (kdembler) [inline packages/atlas/src/views/global/YppDashboard/YppDashboard.tsx]: Could we maybe move this and `TIERS` to a separate file to improve component's readability? Can be called `YppDashboard.config.ts`

**Review by kdembler — CHANGES_REQUESTED** [review]

File: packages/atlas/src/views/viewer/ViewerLayout.tsx [hunk 12]
```diff
@@ -17,6 +17,7 @@ import { transitions } from '@/styles'
 import { RoutingState } from '@/types/routing'
 import { YppLandingView } from '@/views/global/YppLandingView'
 import { NotificationsView } from '@/views/notifications'
+import { YppDashboard } from '@/views/studio/YppDashboard'
```
Comment (kdembler) [inline packages/atlas/src/views/viewer/ViewerLayout.tsx]:
```suggestion
```

File: packages/atlas/src/views/viewer/ViewerLayout.tsx [hunk 13]
```diff
@@ -45,6 +46,7 @@ const viewerRoutes = [
   { path: relativeRoutes.viewer.member(), element: <MemberView /> },
   { path: relativeRoutes.viewer.nfts(), element: <NftsView /> },
   { path: relativeRoutes.viewer.ypp(), element: <YppLandingView /> },
+  { path: relativeRoutes.viewer.yppDashboard(), element: <YppDashboard /> },
```
Comment (kdembler) [inline packages/atlas/src/views/viewer/ViewerLayout.tsx]:
```suggestion
```

**Review by kdembler — APPROVED** [review]

**Pull request merged**